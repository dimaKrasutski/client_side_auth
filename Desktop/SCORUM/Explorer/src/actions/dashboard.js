import { createActions } from 'redux-actions';

export const {
  getChainCapital,
  setChainCapital,
  loadOperations,
  setDashboardOperations,
  loadBlocks,
  setBlocks,
  loadAccountCount,
  setAccountCount,
} = createActions(
  'GET_CHAIN_CAPITAL',
  'SET_CHAIN_CAPITAL',
  'LOAD_OPERATIONS',
  'SET_DASHBOARD_OPERATIONS',
  'LOAD_BLOCKS',
  'SET_BLOCKS',
  'LOAD_ACCOUNT_COUNT',
  'SET_ACCOUNT_COUNT',
);
