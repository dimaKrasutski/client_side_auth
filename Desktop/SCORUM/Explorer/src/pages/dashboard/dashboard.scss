.dashboard-container {
  .dashboard-transactions-table,
  .dashboard-blocks-table {
    td {
      max-width: 100px;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    }
  }

  .text-align {
    vertical-align: text-bottom;
  }

  .btn {
    border-radius: 4px;
  }
}
