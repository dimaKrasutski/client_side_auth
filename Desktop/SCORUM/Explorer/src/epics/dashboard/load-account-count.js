import { ofType } from 'redux-observable';
import { mergeMap } from 'rxjs/operators/mergeMap';
import { map } from 'rxjs/operators/map';
import { loadAccountCount, setAccountCount } from '../../actions/dashboard';
import { getAccountCount } from '../../api/scorum';
import { catchGlobalErrorWith } from '../common-operators';
import { UNDEFINED_ERROR_ID } from '../../constants';

export const loadAccountCountEpic = action$ => action$.pipe(
  ofType(loadAccountCount),
  mergeMap(() => getAccountCount()),
  map(setAccountCount),
  catchGlobalErrorWith(UNDEFINED_ERROR_ID),
);
