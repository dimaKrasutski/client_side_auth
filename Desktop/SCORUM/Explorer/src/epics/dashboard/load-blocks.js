import { ofType } from 'redux-observable';
import { mergeMap } from 'rxjs/operators/mergeMap';
import { map } from 'rxjs/operators/map';
import { compose, reverse } from 'ramda';
import { loadBlocks, setBlocks } from '../../actions/dashboard';
import { getBlocksHistory } from '../../api/scorum';
import { catchGlobalErrorWith } from '../common-operators';
import { UNDEFINED_ERROR_ID } from '../../constants';

export const loadBlocksEpic = action$ => action$.pipe(
  ofType(loadBlocks),
  mergeMap(() => getBlocksHistory(-1, 5)),
  map(compose(
    setBlocks,
    reverse,
  )),
  catchGlobalErrorWith(UNDEFINED_ERROR_ID),
);
