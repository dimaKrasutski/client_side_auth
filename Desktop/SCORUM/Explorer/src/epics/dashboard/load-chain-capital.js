import { ofType } from 'redux-observable';
import { mergeMap } from 'rxjs/operators/mergeMap';
import { map } from 'rxjs/operators/map';
import { getChainCapital as getChainCapitalAction, setChainCapital } from '../../actions/dashboard';
import { getChainCapital } from '../../api/scorum';
import { catchGlobalErrorWith } from '../common-operators';
import { UNDEFINED_ERROR_ID } from '../../constants';

export const loadChainCapitalEpic = action$ => action$.pipe(
  ofType(getChainCapitalAction),
  mergeMap(() => getChainCapital()),
  map(setChainCapital),
  catchGlobalErrorWith(UNDEFINED_ERROR_ID),
);
