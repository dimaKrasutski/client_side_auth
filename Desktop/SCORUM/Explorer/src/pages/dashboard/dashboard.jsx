import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Container from 'reactstrap/lib/Container';
import { FormattedMessage } from 'react-intl';
import { compose } from 'ramda';
import { Redirect } from 'react-router-dom';
import Row from 'reactstrap/lib/Row';
import Col from 'reactstrap/lib/Col';
import Table from 'reactstrap/lib/Table';
import Input from 'reactstrap/lib/Input';
import { IconScr } from 'theme/components/icons/icon-scr/icon-scr';
import { IconSp } from 'theme/components/icons/icon-sp/icon-sp';
import { wrapRoute2Locale } from 'theme/helpers/localization';
import { Main } from '../main';
import { BalanceCard } from '../../components/balance-card/balance-card';
import { BalanceCardBody } from '../../components/balance-card/balance-card-body';
import { DashboardButton } from '../../components/dashboard-button/dashboard-button';
import { loadOperations, loadBlocks, loadAccountCount, getChainCapital } from '../../actions/dashboard';
import { Block } from './block/block';
import { Transaction } from './transaction/transaction';
import './dashboard.scss';

export class DashboardUI extends Component {
  static propTypes = {
    locale: PropTypes.string.isRequired,
    amountSCR: PropTypes.number.isRequired,
    amountSP: PropTypes.number.isRequired,
    blocksCount: PropTypes.number.isRequired,
    operationsCount: PropTypes.number,
    accountCount: PropTypes.number.isRequired,
    blocks: PropTypes.arrayOf().isRequired,
    operations: PropTypes.arrayOf().isRequired,
    preloadChainCapital: PropTypes.func.isRequired,
    loadOperationsAction: PropTypes.func.isRequired,
    loadBlocksAction: PropTypes.func.isRequired,
    loadAccountCountAction: PropTypes.func.isRequired,
  };

  static defaultProps = {
    operationsCount: 0,
  };

  state = {
    redirectToBlocks: false,
    redirectToTransactions: false,
  };

  componentWillMount() {
    const {
      loadOperationsAction, loadBlocksAction, loadAccountCountAction, preloadChainCapital,
    } = this.props;
    preloadChainCapital();
    loadOperationsAction();
    loadBlocksAction();
    loadAccountCountAction();
  }

  goToBlocksPage = () => {
    this.setState({ redirectToBlocks: true });
  };

  goToTransactionsPage = () => {
    this.setState({ redirectToTransactions: true });
  };

  render() {
    const {
      locale,
      amountSCR,
      amountSP,
      blocks,
      operations,
      blocksCount,
      operationsCount,
      accountCount,
    } = this.props;

    const { redirectToBlocks, redirectToTransactions } = this.state;

    if (redirectToBlocks) {
      return <Redirect push to={wrapRoute2Locale('/blocks', locale)} />;
    }

    if (redirectToTransactions) {
      return <Redirect push to={wrapRoute2Locale('/transactions', locale)} />;
    }

    return (
      <Main>
        <Container className="pt-6 dashboard-container">
          <Row>
            <Col xs="4">
              <BalanceCard classNames="card-left" innerClassName="pl-3">
                <BalanceCardBody
                  title="explorer.dashboard.balance-card.total-scr"
                  icon={<IconScr />}
                  balance={amountSCR}
                  trades={1}
                  currency="USD"
                  locale={locale}
                />
              </BalanceCard>
            </Col>
            <Col xs="4">
              <BalanceCard classNames="card-left" innerClassName="pl-3">
                <BalanceCardBody
                  title="explorer.dashboard.balance-card.total-sp"
                  icon={<IconSp />}
                  balance={amountSP}
                  trades={1}
                  currency="USD"
                  locale={locale}
                />
              </BalanceCard>
            </Col>
            <Col xs="4">
              <BalanceCard classNames="card-left h-100" innerClassName="h-100 pl-3">
                <span>
                  <FormattedMessage id="explorer.dashboard.balance-card.total-users" />
                </span>
                <h2 className="mt-1_5 mb-2">{accountCount}</h2>
              </BalanceCard>
            </Col>
          </Row>
          <Row className="mt-6">
            <Col xs="6">
              <Input type="text" plaintext className="mb-0">
                <span className="h3 mr-1">
                  {operationsCount}
                </span>
                <span className="font-weight-normal text-align">
                  <FormattedMessage id="explorer.dashboard.total-transactions" />
                </span>
              </Input>
              <Table className="dashboard-transactions-table">
                <thead className="thead-light">
                  <tr>
                    <th>
                      <FormattedMessage id="explorer.dashboard.transactions.hash" />
                    </th>
                    <th>
                      <FormattedMessage id="explorer.dashboard.transactions.date" />
                    </th>
                    <th>
                      <FormattedMessage id="explorer.dashboard.transactions.type" />
                    </th>
                  </tr>
                </thead>
                <tbody>
                  {
                    operations.map(operation => (
                      <Transaction key={operation[1].trx_id} locale={locale} transaction={operation[1]} />
                    ))
                  }
                </tbody>
              </Table>
              <DashboardButton text="explorer.dashboard.all-transactions" onClick={this.goToTransactionsPage} />
            </Col>
            <Col xs="6">
              <Input type="text" plaintext className="mb-0">
                <span className="h3 mr-1">
                  {blocksCount}
                </span>
                <span className="font-weight-normal text-align">
                  <FormattedMessage id="explorer.dashboard.total-blocks" />
                </span>
              </Input>
              <Table className="dashboard-blocks-table">
                <thead className="thead-light">
                  <tr>
                    <th>
                      <FormattedMessage id="explorer.dashboard.blocks.height" />
                    </th>
                    <th>
                      <FormattedMessage id="explorer.dashboard.blocks.date" />
                    </th>
                    <th>
                      <FormattedMessage id="explorer.dashboard.blocks.hash" />
                    </th>
                  </tr>
                </thead>
                <tbody>
                  {
                    blocks.map(block => (<Block key={block[1].block_id} locale={locale} block={block} />))
                  }
                </tbody>
              </Table>
              <DashboardButton text="explorer.dashboard.all-blocks" onClick={this.goToBlocksPage} />
            </Col>
          </Row>
        </Container>
      </Main>
    );
  }
}

export const mapStateToProps = ({ intl, dashboard }) => ({
  locale: intl.localeFull,
  amountSCR: dashboard.amountSCR,
  amountSP: dashboard.amountSP,
  blocksCount: dashboard.blocksCount,
  operationsCount: dashboard.operationsCount,
  blocks: dashboard.blocks,
  operations: dashboard.operations,
  accountCount: dashboard.accountCount,
});

export const mapDispatchToProps = dispatch => ({
  preloadChainCapital: compose(dispatch, getChainCapital),
  loadOperationsAction: compose(dispatch, loadOperations),
  loadBlocksAction: compose(dispatch, loadBlocks),
  loadAccountCountAction: compose(dispatch, loadAccountCount),
});

export const Dashboard = connect(mapStateToProps, mapDispatchToProps)(DashboardUI);
