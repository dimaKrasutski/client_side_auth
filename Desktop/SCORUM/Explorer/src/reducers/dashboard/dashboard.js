import { handleActions } from 'redux-actions';
import initialState from './initial-state';
import { setBlocks, setDashboardOperations, setAccountCount, setChainCapital } from '../../actions/dashboard';

export default handleActions(
  {
    [setChainCapital]: (state, { payload: { circulating_scr: totalScr, circulating_sp: totalSP } }) => ({
      ...state,
      amountSCR: totalScr,
      amountSP: totalSP,
    }),
    [setBlocks]: (state, { payload: blocks }) => ({
      ...state,
      blocks,
      blocksCount: blocks[0][0],
    }),
    [setDashboardOperations]: (state, { payload: operations }) => ({
      ...state,
      operations,
      operationsCount: operations[0][0],
    }),
    [setAccountCount]: (state, { payload: accountCount }) => ({
      ...state,
      accountCount,
    }),
  },
  initialState,
);
