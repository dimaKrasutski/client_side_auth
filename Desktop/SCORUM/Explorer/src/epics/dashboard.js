import { combineEpics } from 'redux-observable';
import { loadBlocksEpic } from './dashboard/load-blocks';
import { loadOperationsEpic } from './dashboard/load-operations';
import { loadAccountCountEpic } from './dashboard/load-account-count';
import { loadChainCapitalEpic } from './dashboard/load-chain-capital';

export const dashboardEpics = combineEpics(
  loadOperationsEpic,
  loadBlocksEpic,
  loadAccountCountEpic,
  loadChainCapitalEpic,
);
