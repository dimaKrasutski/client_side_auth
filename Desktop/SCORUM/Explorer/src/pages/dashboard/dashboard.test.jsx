import React from 'react';
import { DashboardUI } from './dashboard';

describe('<DashboardUI />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow((<DashboardUI
      locale="en-us"
      preloadChainCapital={jest.fn()}
      loadOperationsAction={jest.fn()}
      loadBlocksAction={jest.fn()}
      loadAccountCountAction={jest.fn()}
      amountSCR={0}
      amountSP={0}
      blocks={[]}
      operations={[]}
      blocksCount="1"
    />));
    expect(wrapper).toMatchSnapshot();
  });
});
