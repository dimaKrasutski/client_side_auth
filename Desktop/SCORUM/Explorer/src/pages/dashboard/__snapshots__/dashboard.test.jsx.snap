// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardUI /> should match snapshot 1`] = `
<Connect(MainUI)>
  <Container
    className="pt-6 dashboard-container"
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="4"
      >
        <BalanceCard
          classNames="card-left"
          innerClassName="pl-3"
        >
          <BalanceCardBody
            balance={0}
            currency="USD"
            icon={<IconScr />}
            locale="en-us"
            title="explorer.dashboard.balance-card.total-scr"
            trades={1}
          />
        </BalanceCard>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="4"
      >
        <BalanceCard
          classNames="card-left"
          innerClassName="pl-3"
        >
          <BalanceCardBody
            balance={0}
            currency="USD"
            icon={<IconSp />}
            locale="en-us"
            title="explorer.dashboard.balance-card.total-sp"
            trades={1}
          />
        </BalanceCard>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="4"
      >
        <BalanceCard
          classNames="card-left h-100"
          innerClassName="h-100 pl-3"
        >
          <span>
            <FormattedMessage
              id="explorer.dashboard.balance-card.total-users"
              values={Object {}}
            />
          </span>
          <h2
            className="mt-1_5 mb-2"
          />
        </BalanceCard>
      </Col>
    </Row>
    <Row
      className="mt-6"
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <Input
          className="mb-0"
          plaintext={true}
          type="text"
        >
          <span
            className="h3 mr-1"
          >
            0
          </span>
          <span
            className="font-weight-normal text-align"
          >
            <FormattedMessage
              id="explorer.dashboard.total-transactions"
              values={Object {}}
            />
          </span>
        </Input>
        <Table
          className="dashboard-transactions-table"
          responsiveTag="div"
          tag="table"
        >
          <thead
            className="thead-light"
          >
            <tr>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.transactions.hash"
                  values={Object {}}
                />
              </th>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.transactions.date"
                  values={Object {}}
                />
              </th>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.transactions.type"
                  values={Object {}}
                />
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
        <DashboardButton
          onClick={[Function]}
          text="explorer.dashboard.all-transactions"
        />
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <Input
          className="mb-0"
          plaintext={true}
          type="text"
        >
          <span
            className="h3 mr-1"
          >
            1
          </span>
          <span
            className="font-weight-normal text-align"
          >
            <FormattedMessage
              id="explorer.dashboard.total-blocks"
              values={Object {}}
            />
          </span>
        </Input>
        <Table
          className="dashboard-blocks-table"
          responsiveTag="div"
          tag="table"
        >
          <thead
            className="thead-light"
          >
            <tr>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.blocks.height"
                  values={Object {}}
                />
              </th>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.blocks.date"
                  values={Object {}}
                />
              </th>
              <th>
                <FormattedMessage
                  id="explorer.dashboard.blocks.hash"
                  values={Object {}}
                />
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
        <DashboardButton
          onClick={[Function]}
          text="explorer.dashboard.all-blocks"
        />
      </Col>
    </Row>
  </Container>
</Connect(MainUI)>
`;
